msteams-platform/bots/how-to/conversations/bot-copilot-handoff.md

```diff
@@ -0,0 +1,178 @@
+
+   > [!NOTE]
+   > Don't send any payload with this response as it doesn't render in the chat window. The responses based on the continuation token must be sent to the user separately.
+
+   Here's an example of how to handle the invoke call in `searchApp.ts`:
+
+   ```typescript
+    public async onInvokeActivity(context: TurnContext): Promise<InvokeResponse> {
+        try {
+          switch (context.activity.name) {
+            case "handoff/action": {
+              // TODO: Save continuation token and use it to process final response to user later
+             return {status: 200}; // return just the http status
+            }
+            case "composeExtension/query":
+              return {
+                status: 200,
+                body: await this.handleTeamsMessagingExtensionQuery(
+                  context,
+                  context.activity.value
+                ),
+              };
+            default:
+              return {
+                status: 200,
+                body: `Unknown invoke activity handled as default- ${context.activity.name}`,
+              };
+          }
+        } catch (err) {
+          console.log(`Error in onInvokeActivity: ${err}`);
+          return {
+            status: 500,
+            body: `Invoke activity received- ${context.activity.name}`,
+          };
+        }
+      }
+   ```
+
+   When the bot receives the invoke call, `context.activity.value.continuation` contains the `continuationToken` that was set in the deep link URL.
+
+## Best practices
+
+* It’s important for the bot to quickly notify users when they're directed to the bot chat after selecting the action button, as there's no visual indication. This helps manage expectations as there might be a delay before the bot returns a response due to network latency and processing time. For instance, the bot can send a series of activities to keep the user informed of the progress:
+
+    ```typescript
+    await context.sendActivities([
+      {
+        type: ActivityTypes.Message,
+        text: "Continuing conversation from Copilot...",
+      },
+      { type: ActivityTypes.Typing },
+      { type: "delay", value: 1000 },
+      {
+        type: ActivityTypes.Message,
+        text: `Fetching more details using the continuation token passed: ${continuationToken}`,
+      },
+      { type: ActivityTypes.Typing },
+      { type: "delay", value: 4000 },
+      {
+        type: ActivityTypes.Message,
+        text: `Handoff successful!`,
+      }
+    ]);
+    
+    ```
+
+* For a smooth handoff process, it’s important to manage the continuation token effectively. Keep the token’s lifetime short and ensure it doesn't process more than once. After a token is used, remove it from storage to prevent it from being used again. If the same token comes up, let the user know they need to start a new conversation with the bot because the handoff from Copilot can't continue.
```

Comment:
Can we explain why it's "important" to have a short lifetime, and ensure that the token is only redeemed once? If the action is idempotent why is this necessary?

Reply:
As discussed with Gino and prem, updated a generic guidance to state that developers must handle the token expiry and multiple reuses of the token.

"We recommended that you manage the lifecycle of the continuation token to ensure it expires after a reasonable period and also handle scenarios where the continuation token request is replayed by the user."